src/fast_check/mod.rs

```diff
@@ -72,6 +72,10 @@ pub enum FastCheckDiagnostic {
   UnsupportedTsNamespaceExport { range: Range },
   #[error("Using declarations are not supported in the public API.")]
   UnsupportedUsing { range: Range },
+  #[error("Referencing a JavaScript file from a TypeScript file is not supported. Add a declaration file or convert to TypeScript.")]
+  UnsupportedNestedJavaScript { range: Range },
+  #[error("JavaScript entrypoints are not supported for fast check.")]
+  UnsupportedJavaScriptEntrypoint { range: Range },
```

Comment:
This is just a diagnostic and not a hard error. In the CLI we can catch this diagnostic and then display a warning that we're publishing without fast check support.